### DISCUSSION TITLE
Warning: Winget is not installed or outdated.

### DISCUSSION THREAD

**Discussion by tassoman** [Raphire/Win11Debloat#75] [Q&A] [answered]
I got this warning message:

> Winget is not installed or outdated. This may prevent Win11Debloat from removing certain apps.

Any idea how to improve?

**Comment by Raphire** [comment, marked as answer]
Win11Debloat requires Winget version 1.4 or later to uninstall both Edge and OneDrive. Windows includes Winget by default, but the pre-installed version may be older than the required v1.4. If you don't care about removing these applications you can safely ignore this warning. 

You can check whether Winget is installed and it's version by running the command `winget -v` in powershell.

If it's installed it will return it's version number, which looks something like this:
![image](https://github.com/Raphire/Win11Debloat/assets/9938813/0b06a943-04f4-42cf-a381-20169ea68984)

Winget can be installed and updated via the Microsoft Store. This is the specific application: https://apps.microsoft.com/detail/9nblggh4nns1

**Reply by tassoman** [reply]
Thanks for this detailed reply!
Unfortunately, my system is completely missing `winget`, also MS Store is missing and I can't download that app.

```powershell
+ winget -v
+ ~~~~~~
    + CategoryInfo          : ObjectNotFound: (winget:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```

**Comment by brightoven** [comment]
Got the same warning when running the W11Debloat on a fresh installed W11pro 23H2. As I didn't want to connect it to the Internet, I downloaded the current [Winget](https://aka.ms/getwinget) and the [WinUI package](https://github.com/microsoft/microsoft-ui-xaml/releases/download/v2.8.6/Microsoft.UI.Xaml.2.8.x64.appx) via another machine and then installed both using Add-AppxPackage.
See [Install WinGet on Windows Sandbox](https://learn.microsoft.com/en-us/windows/package-manager/winget/#install-winget-on-windows-sandbox) for more details. (I did not need to install the VCLibs on my 23H2.)

**Comment by IT-CMS24** [comment]
I am seeing the Winget warning, but the machine shows 1.12.460.  Not sure if that is there by default or am I missing something?

**Reply by Raphire** [reply]
You shouldn't be seeing it, I am running the same version. Are you using the latest version of the script?

**Reply by IT-CMS24** [reply]
I am running the newest version with the new GUI that shows the rocket ship.
and winget version is 1.12.460 (via ps "winget -v")
![Debloat](https://github.com/user-attachments/assets/15d2b96c-3302-4608-86c8-8770aa4f882f)

**Reply by Raphire** [reply]
That's strange. Could you try running the winget command in an elevated powershell window if you haven't already? 

E.g. launch terminal by right-clicking on the windows icon on the taskbar and click Terminal (Admin) 
<img width="230" height="621" alt="image" src="https://github.com/user-attachments/assets/d80cc516-1c4d-4680-95b9-07bbb76e047f" />

**Reply by IT-CMS24** [reply]
Running it as admin shows that it is running v1.12.460.  Same as before.

**Reply by Raphire** [reply]
Thanks for confirming, hard to say why it's giving this warning then. I'll do some more testing on my end.

**Reply by Leni-Vienne** [reply]
I have the exact same issue, where the microsoft store / .msixbundle says a recent version is installed and `winget -v` says it's an old one. Pretty insane stuff. and winget can't be removed or re installed. Though not a Win11Debloat issue
<img width="1290" height="492" alt="image" src="https://github.com/user-attachments/assets/101684cb-d12d-404b-83eb-81e80ae2b0be" />

<img width="1177" height="237" alt="image" src="https://github.com/user-attachments/assets/00f52b42-8f44-447d-80d6-fdb37741f507" />

**Reply by Raphire** [reply]
> I have the exact same issue, where the microsoft store / .msixbundle says a recent version is installed and `winget -v` says it's an old one. Pretty insane stuff. and winget can't be removed or re installed. Though not a Win11Debloat issue <img alt="image" width="1290" height="492" src="https://private-user-images.githubusercontent.com/71212417/552056442-101684cb-d12d-404b-83eb-81e80ae2b0be.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzE1MDA2ODYsIm5iZiI6MTc3MTUwMDM4NiwicGF0aCI6Ii83MTIxMjQxNy81NTIwNTY0NDItMTAxNjg0Y2ItZDEyZC00MDRiLTgzZWItODFlODBhZTJiMGJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjE5VDExMjYyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3OTA5OWRiZjYxNGU0NmEwOGQ2MzQ4MDMwMmQyNzcwYzA5MjIwMzIwYmE0NjNkNzlmNGM2ODFmZDkwMjFhMTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zxRN-ZU6ElDgaXllEaW70C-wQla0h7ONOdQP399agKE">
> 
> <img alt="image" width="1177" height="237" src="https://private-user-images.githubusercontent.com/71212417/552058381-00f52b42-8f44-447d-80d6-fdb37741f507.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzE1MDA2ODYsIm5iZiI6MTc3MTUwMDM4NiwicGF0aCI6Ii83MTIxMjQxNy81NTIwNTgzODEtMDBmNTJiNDItOGY0NC00NDdkLTgwZDYtZmRiMzc3NDFmNTA3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjE5VDExMjYyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3Yzg0OWE0M2QyZGI5N2Y2YmQ5Y2Y5YmZlMjJmY2YwM2I5OTY4NzA1YTcyNGY4NGQ2MDdhOTI1ZmY2MDU2ODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WXPCsrVrpn1KqF9ff7IZxIhfBcAe0kBZ53mX8D0gzuU">

Interesting, the output of 1.12 should still not show a warning though, the script checks if the version is higher than 1.4.

**Comment by IT-CMS24** [comment]
Thank you.  Much appreciated.  Do you happen to have any tools to convert the script into something that can be deployed via Intune for Autopilot devices?

**Reply by Raphire** [reply]
I have documented some automation examples here: https://github.com/Raphire/Win11Debloat/wiki/Automation

It might work best to create your own script that calls Win11Debloat. You can change the script behaviour by supplying the respective parameters, which are documented [here](https://github.com/Raphire/Win11Debloat/wiki/Command%E2%80%90line-Interface). Should all work without user intervention.

**Comment by Raphire** [comment]
@IT-CMS24 @Leni-Vienne I have changed how the script checks the installation status of winget, could you please try the script again to see if this fixes the issue? If not, could you see if your log contains the line `Unable to determine if WinGet is installed, winget command failed:`

**Comment by Leni-Vienne** [comment]
Actually the warning was shown because for some reason my PATH variable pointed to the wrong drive, your script ended up working perfectly after fixing that.

I just tested the new script after modifying the PATH back to nowhere and the warning `Winget is not installed or outdated` does appear. But yeah in my case it was a path issue more than an outdated version issue

**Reply by Raphire** [reply]
I see, thanks for the detailed reply! Glad you figured it out.
